Description: Fix SELinux build failure
Origin: other, https://bugzilla.mindrot.org/attachment.cgi?id=1991&action=diff
Author: Leonardo Chiqitto <leonardo@ngdn.org>
Bug: https://bugzilla.mindrot.org/show_bug.cgi?id=1851
Last-Update: 2011-02-05

Index: b/openbsd-compat/port-linux.c
===================================================================
--- a/openbsd-compat/port-linux.c
+++ b/openbsd-compat/port-linux.c
@@ -226,7 +226,7 @@
 
 	if (!ssh_selinux_enabled())
 		return;
-	if (path == NULL)
+	if (path == NULL) {
 		setfscreatecon(NULL);
 		return;
 	}
