From 727d51f30918f6635f06694f71f4318a6038296d Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Sun, 9 Feb 2014 16:10:17 +0000
Subject: Support synchronisation with service supervisor using SIGSTOP

Forwarded: no
Last-Update: 2013-09-14

Patch-Name: sigstop.patch
---
 sshd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sshd.c b/sshd.c
index 63b9357..fd7f182 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1909,6 +1909,10 @@ main(int ac, char **av)
 			}
 		}
 
+		if (getenv("SSH_SIGSTOP"))
+			/* Tell service supervisor that we are ready. */
+			kill(getpid(), SIGSTOP);
+
 		/* Accept a connection and return in a forked child */
 		server_accept_loop(&sock_in, &sock_out,
 		    &newsock, config_s);
